pio-scripts/decoder.py

```
@@ -0,0 +1,94 @@
+            raise Exception("addr2line for ESP not found in {} - {} does not exist.\nUse --toolchain to point to "
+                            "your toolchain folder.".format(self.toolchain_path, self.addr2line_path))
+
+        self.elf_path = elf_path
+        if not os.path.exists(self.elf_path):
+            raise Exception("ELF file not found: '{}'".format(self.elf_path))
+
+    def parse_text(self, text):
+        m = re.search('Backtrace: (.*)', text)
+        if m:
+            print ("Stack trace:")
+            for l in self.parse_stack(m.group(1)):
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)